### PR TITLE
Implementing InferenceEngine::CNNNetwork::setBatchSize() in IE C API

### PR DESCRIPTION
**Pull request by mmakridi**

### CODE REVIEW

File: inference-engine/ie_bridges/c/include/c_api/ie_c_api.h [hunk 1]
```diff
@@ -709,6 +709,16 @@ INFERENCE_ENGINE_C_API(IE_NODISCARD IEStatusCode) ie_network_get_input_precision
...
 
+/**
+ * @brief Sets new batch size limit for the network.
+ * This function should be called before loading the network to the device.
+ * @ingroup Network
+ * @param network A pointer to ie_network_t instance.
+ * @param batch_size New batch size to be used.
+ * @return Status code of the operation: OK(0) for success.
+ */
+INFERENCE_ENGINE_C_API(IE_NODISCARD IEStatusCode) ie_network_set_batch_size(ie_network_t *network, size_t batch_size);
```
Comment (ilyachur) [inline inference-engine/ie_bridges/c/include/c_api/ie_c_api.h:720]: Can we add new tests for this method in order to check that it works as expected?

Comment (mmakridi) [reply inference-engine/ie_bridges/c/include/c_api/ie_c_api.h:720]: Added tests in a manner similar to `ie_infer_request_set_batch`. May you please explain, how can I validate these tests? Do they run automatically in CI for this commit?

**Review by ilyachur — APPROVED** [review]

File: inference-engine/ie_bridges/c/include/c_api/ie_c_api.h (hunk 1 above)
Comment (ilyachur) [reply inference-engine/ie_bridges/c/include/c_api/ie_c_api.h:720]: As I know, our precomit should include these tests.

**Comment by ilyachur** [comment]
Please fix CI

**Pull request closed without merging**